core/src/main/java/org/mule/management/stats/ComponentStatistics.java

```diff
@@ -156,7 +156,7 @@ public synchronized void addExecutionBranchTime(boolean first, long branch, long
         {
             maxExecutionTime = effectiveTotal;
         }
-        averageExecutionTime = Math.round(totalExecTime / executedEvent);
+        averageExecutionTime = executedEvent > 0 ? Math.round(totalExecTime / executedEvent) : 0;
```

Comment:
I ran into tests failing because of this. Could extract it to a separate issue but I don't think it's worth it